explorer/info.go

```diff
@@ -0,0 +1,82 @@
+package explorer
+
+import (
+	"time"
+
+	"go.sia.tech/core/types"
+)
+
+// BlockFacts returns a bunch of statistics about the consensus set as they
+// were at a specific block.
+type BlockFacts struct {
+	Index             types.ChainIndex
+	TotalWork         types.Work
+	Difficulty        types.Work
+	OakWork           types.Work
+	OakTime           time.Duration
+	GenesisTimestamp  time.Time
+	SiafundPool       types.Currency
+	FoundationAddress types.Address
+
+	// Transaction type counts.
+	SpentSiacoinsCount         uint64
+	SpentSiafundsCount         uint64
+	NewSiacoinsCount           uint64
+	NewSiafundsCount           uint64
+	NewFileContractsCount      uint64
+	RevisedFileContractsCount  uint64
+	ResolvedFileContractsCount uint64
+
+	// Facts about file contracts.
+	ActiveContractCost  types.Currency
+	ActiveContractCount uint64
+	ActiveContractSize  uint64
+	TotalContractCost   types.Currency
+	TotalContractSize   uint64
+	TotalRevisionVolume uint64
```

Comment:
I would rename this struct `ChainStats`, and remove the fields taken from `ValidationContext` -- they can be handled separately